test/test_sparse_csr.py

```diff
@@ -77,7 +75,47 @@ def test_factory_size_check(self, device, dtype):
             torch.sparse_csr_tensor(torch.tensor(crow_indices), torch.tensor(col_indices), torch.tensor([0, 0, 0, 0, 0]),
                                     size, dtype=dtype, device=device)
 
-    @onlyCPU
+    @onlyCUDA
+    @dtypes(*torch.testing.get_all_dtypes(include_bool=False, include_half=False,
+                                          include_bfloat16=False, include_complex=False))
+    def test_factory_device_type_inference(self, device, dtype):
+        sparse = torch.sparse_csr_tensor([0, 2, 4],
+                                         [0, 1, 0, 1],
+                                         [1, 2, 3, 4],
+                                         size=(2, 10),
+                                         dtype=dtype,
+                                         device=device)
+
+        self.assertEqual((2, 10), sparse.shape)
+        self.assertEqual(torch.tensor([0, 2, 4], dtype=torch.int32, device=device), sparse.crow_indices())
+        self.assertEqual(torch.tensor([0, 1, 0, 1], dtype=torch.int32, device=device), sparse.col_indices())
+        self.assertEqual(torch.tensor([1, 2, 3, 4], dtype=dtype, device=device), sparse.values())
```

Comment:
It runs with `device=='cpu'` as well as you can see here.
https://github.com/pytorch/pytorch/blob/6c6184f3d38fae4756cfbb2fa2cdfb4ae3db06c4/test/test_sparse_csr.py#L96
 It generates a different kind of combinations among ('cpu', 'cuda', None) to test device type inference.

Reply:
See https://github.com/pytorch/pytorch/blob/6c6184f3d38fae4756cfbb2fa2cdfb4ae3db06c4/test/test_sparse_csr.py#L78 doesn't this disable CPU runs?

Reply:
It doesn't disable CPU. If @onlyCUDA is enabled, the test was designed to work with CUDA because the functions you are using was compiled with CUDA, so that means that it was compiled for CPU as well.  To probe that I run the test with @onlyCUDA  enabled and @onlyCPU enabled.   So basically it uses onlyCUDA to not run twice.

Reply:
I can see the point of avoiding running the tests twice. However, when pytorch is built without CUDA support, the tests are not run, AFAIRFC.